### DISCUSSION TITLE
Enum definitions in IDLs

### DISCUSSION THREAD

**Discussion by anwaymukherjee** [eProsima/Fast-DDS-Gen#286] [Support]
Quick question,

if an IDL has the following
format #1
enum EnumName
{
  NOTINITIALIZED = 0, 
  NORMAL = 1
};

Can we use it as is, or do we have to format it like this ?
format #2
enum EnumName
{
  @value(0) NOTINITIALIZED, 
  @value(1) NORMAL
};

Because I am getting an error when I pass the following format #1 to fastddsgen
Also, is there any plan to use enum class in the cxx impl ?  That would possibly avoid a lot of compilation conflicts

**Comment by JLBuenoLopez** [comment]
@anwaymukherjee 

I am afraid that [IDL specification](https://www.omg.org/spec/IDL/4.2/About-IDL) does not allow the first format. The only way to set specific values in an enumeration is the second format. However, I am afraid that Fast DDS-Gen does not support yet the builtin `@value` annotation. You might contact [eProsima commercial support team](https://github.com/eProsima/Fast-DDS?tab=readme-ov-file#getting-help) for options to accelerate its support.

Concerning the `enum class` implementation, it is something to be expected in the next major release for Fast DDS-Gen. #271 introduced this feature, but it is still in the development branch and not merged into master. Stay tuned!

Finally, I am moving this conversation to the Support discussion forum as it is not a proper bug in the library as per [Fast DDS Contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support)
